### ISSUE TITLE
Segmentation fault in dispatch() while decoding a message  on s390

### ISSUE THREAD

**Issue by pantras** [protocolbuffers/protobuf#8716]
**What version of protobuf and what language are you using?**
Version: 3.11.0 (google-protobuf gem version)
Language: Ruby

**What operating system (Linux, Windows, ...) and version?**
Red Hat Enterprise Linux Server release 7.9 (Maipo)

**What runtime / compiler are you using (e.g., python version or gcc version)**
ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [s390x-linux]

**What did you do?**
I decode a message that I received. 
I embedded the message in a small script, see  below.

**What did you expect to see**
I expected  to get a Hash  object containing the decoded message.

**What did you see instead?**
A crash symbol in the dispatch  function. 

gdb stack:
```
(gdb) where
#0  0x000003ffa4841370 in raise () from /lib64/libc.so.6
#1  0x000003ffa4842bf8 in abort () from /lib64/libc.so.6
#2  0x000003ffa4ee4e1a in rb_bug_for_fatal_signal () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#3  0x000003ffa5035530 in sigsegv () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#4  <signal handler called>
#5  0x000003ffa11413da in dispatch ()
   from /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
#6  0x000003ffa1142446 in run_decoder_vm ()
   from /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
#7  0x000003ffa113cde0 in upb_bufsrc_putbuf ()
   from /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
#8  0x000003ffa1124508 in Message_decode ()
   from /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
#9  0x000003ffa50b90c6 in vm_call_cfunc () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#10 0x000003ffa50d7bda in vm_call_method () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#11 0x000003ffa50c6524 in vm_exec_core () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#12 0x000003ffa50ccb22 in rb_vm_exec () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#13 0x000003ffa4eebd50 in rb_ec_exec_node () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#14 0x000003ffa4ef1b64 in ruby_run_node () from /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7
#15 0x0000000080000a38 in main (argc=2, argv=0x3ffe257e5c8) at ./main.c:50
```
Output of the script
```
[root@twenty1 bin]# ruby dispatch_crash.rb
dispatch_crash.rb:49: [BUG] Segmentation fault at 0xb8c49ae800000000
ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [s390x-linux]

-- Control frame information -----------------------------------------------
c:0003 p:---- s:0014 e:000013 CFUNC  :decode
c:0002 p:0130 s:0009 E:001270 EVAL   dispatch_crash.rb:49 [FINISH]
c:0001 p:0000 s:0003 E:0021d0 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
dispatch_crash.rb:49:in `<main>'
dispatch_crash.rb:49:in `decode'

-- C level backtrace information -------------------------------------------
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(rb_vm_bugreport+0xfc) [0x3ffba060f94]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(rb_bug_for_fatal_signal+0xae) [0x3ffb9e64dfe]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(sigsegv+0x0) [0x3ffb9fb5530]
[0xb891eb78]
[0x3ffb60c13da]
[0x3ffb60c2446]
[0x3ffb60bcde0]
[0x3ffb60a4508]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(vm_call_cfunc+0x166) [0x3ffba0390c6]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(vm_call_method+0xd2) [0x3ffba057bda]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(vm_exec_core+0x174) [0x3ffba046524]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(rb_vm_exec+0x77a) [0x3ffba04cb22]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(rb_ec_exec_node+0xe8) [0x3ffb9e6bd50]
/opt/rbenv/versions/2.7.3/lib/libruby.so.2.7(ruby_run_node+0x5c) [0x3ffb9e71b64]
/opt/rbenv/versions/2.7.3/bin/ruby(main+0x70) [0x80000a38] ./main.c:50

-- Other runtime information -----------------------------------------------

* Loaded script: dispatch_crash.rb

* Loaded features:

    0 enumerator.so
    1 thread.rb
    2 rational.so
    3 complex.so
    4 ruby2_keywords.rb
    5 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/encdb.so
    6 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/trans/transdb.so
    7 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/rbconfig.rb
    8 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/compatibility.rb
    9 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/defaults.rb
   10 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/deprecate.rb
   11 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/errors.rb
   12 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/version.rb
   13 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/requirement.rb
   14 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/platform.rb
   15 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/basic_specification.rb
   16 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/stub_specification.rb
   17 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/util.rb
   18 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/text.rb
   19 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/user_interaction.rb
   20 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/specification_policy.rb
   21 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/util/list.rb
   22 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/specification.rb
   23 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/exceptions.rb
   24 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/bundler_version_finder.rb
   25 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/dependency.rb
   26 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/core_ext/kernel_gem.rb
   27 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/monitor.so
   28 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/monitor.rb
   29 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb
   30 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/core_ext/kernel_warn.rb
   31 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems.rb
   32 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/rubygems/path_support.rb
   33 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/version.rb
   34 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/core_ext/name_error.rb
   35 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/levenshtein.rb
   36 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/jaro_winkler.rb
   37 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checker.rb
   38 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/name_error_checkers/class_name_checker.rb
   39 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/name_error_checkers/variable_name_checker.rb
   40 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/name_error_checkers.rb
   41 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/method_name_checker.rb
   42 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/key_error_checker.rb
   43 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/spell_checkers/null_checker.rb
   44 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/formatters/plain_formatter.rb
   45 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean/tree_spell_checker.rb
   46 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/did_you_mean.rb
   47 /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf/message_exts.rb
   48 /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
   49 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/forwardable/impl.rb
   50 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/forwardable/version.rb
   51 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/forwardable.rb
   52 /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf/repeated_field.rb
   53 /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf.rb
   54 /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/base64.rb

* Process memory map:

80000000-80001000 r-xp 00000000 fd:00 101832986                          /opt/rbenv/versions/2.7.3/bin/ruby
80001000-80002000 r--p 00000000 fd:00 101832986                          /opt/rbenv/versions/2.7.3/bin/ruby
80002000-80003000 rw-p 00001000 fd:00 101832986                          /opt/rbenv/versions/2.7.3/bin/ruby
b8818000-b8839000 rw-p 00000000 00:00 0                                  [heap]
b8839000-b8c6c000 rw-p 00000000 00:00 0                                  [heap]
3ffb5f80000-3ffb5f90000 r-xp 00000000 fd:00 33554510                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
3ffb5f90000-3ffb5f91000 r--p 0000f000 fd:00 33554510                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
3ffb5f91000-3ffb5f92000 rw-p 00010000 fd:00 33554510                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
3ffb6000000-3ffb6025000 r--s 00000000 fd:00 101832986                    /opt/rbenv/versions/2.7.3/bin/ruby
3ffb6080000-3ffb60da000 r-xp 00000000 fd:00 504682728                    /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
3ffb60da000-3ffb60db000 ---p 0005a000 fd:00 504682728                    /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
3ffb60db000-3ffb60dc000 r--p 0005a000 fd:00 504682728                    /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
3ffb60dc000-3ffb60de000 rw-p 0005b000 fd:00 504682728                    /opt/rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/google-protobuf-3.11.0/lib/google/protobuf_c.so
3ffb6100000-3ffb6102000 r-xp 00000000 fd:00 201755106                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/monitor.so
3ffb6102000-3ffb6103000 r--p 00001000 fd:00 201755106                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/monitor.so
3ffb6103000-3ffb6104000 rw-p 00002000 fd:00 201755106                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/monitor.so
3ffb6180000-3ffb6182000 r-xp 00000000 fd:00 302040545                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/trans/transdb.so
3ffb6182000-3ffb6183000 ---p 00002000 fd:00 302040545                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/trans/transdb.so
3ffb6183000-3ffb6184000 r--p 00002000 fd:00 302040545                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/trans/transdb.so
3ffb6184000-3ffb6185000 rw-p 00003000 fd:00 302040545                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/trans/transdb.so
3ffb61b6000-3ffb61b7000 ---p 00000000 00:00 0
3ffb61b7000-3ffb6258000 rw-p 00000000 00:00 0
3ffb6258000-3ffb6259000 ---p 00000000 00:00 0
3ffb6259000-3ffb62fa000 rw-p 00000000 00:00 0
3ffb62fa000-3ffb62fb000 ---p 00000000 00:00 0
3ffb62fb000-3ffb639c000 rw-p 00000000 00:00 0
3ffb639c000-3ffb639d000 ---p 00000000 00:00 0
3ffb639d000-3ffb643e000 rw-p 00000000 00:00 0
3ffb643e000-3ffb643f000 ---p 00000000 00:00 0
3ffb643f000-3ffb64e0000 rw-p 00000000 00:00 0
3ffb64e0000-3ffb64e1000 ---p 00000000 00:00 0
3ffb64e1000-3ffb6582000 rw-p 00000000 00:00 0
3ffb6582000-3ffb6583000 ---p 00000000 00:00 0
3ffb6583000-3ffb6624000 rw-p 00000000 00:00 0
3ffb6624000-3ffb6625000 ---p 00000000 00:00 0
3ffb6625000-3ffb66c6000 rw-p 00000000 00:00 0
3ffb66c6000-3ffb66c7000 ---p 00000000 00:00 0
3ffb66c7000-3ffb6768000 rw-p 00000000 00:00 0
3ffb6768000-3ffb6769000 ---p 00000000 00:00 0
3ffb6769000-3ffb680a000 rw-p 00000000 00:00 0
3ffb680a000-3ffb680b000 ---p 00000000 00:00 0
3ffb680b000-3ffb68ac000 rw-p 00000000 00:00 0
3ffb68ac000-3ffb68ad000 ---p 00000000 00:00 0
3ffb68ad000-3ffb694e000 rw-p 00000000 00:00 0
3ffb694e000-3ffb694f000 ---p 00000000 00:00 0
3ffb694f000-3ffb69f0000 rw-p 00000000 00:00 0
3ffb69f0000-3ffb69f1000 ---p 00000000 00:00 0
3ffb69f1000-3ffb6a92000 rw-p 00000000 00:00 0
3ffb6a92000-3ffb6a93000 ---p 00000000 00:00 0
3ffb6a93000-3ffb6b34000 rw-p 00000000 00:00 0
3ffb6b34000-3ffb6b35000 ---p 00000000 00:00 0
3ffb6b35000-3ffb6bd6000 rw-p 00000000 00:00 0
3ffb6bd6000-3ffb6bd7000 ---p 00000000 00:00 0
3ffb6bd7000-3ffb6c78000 rw-p 00000000 00:00 0
3ffb6c78000-3ffb6c79000 ---p 00000000 00:00 0
3ffb6c79000-3ffb6d1a000 rw-p 00000000 00:00 0
3ffb6d1a000-3ffb6d1b000 ---p 00000000 00:00 0
3ffb6d1b000-3ffb6dbc000 rw-p 00000000 00:00 0
3ffb6dbc000-3ffb6dbd000 ---p 00000000 00:00 0
3ffb6dbd000-3ffb6e5e000 rw-p 00000000 00:00 0
3ffb6e5e000-3ffb6e5f000 ---p 00000000 00:00 0
3ffb6e5f000-3ffb6f00000 rw-p 00000000 00:00 0
3ffb6f00000-3ffb6f01000 ---p 00000000 00:00 0
3ffb6f01000-3ffb6fa2000 rw-p 00000000 00:00 0
3ffb6fa2000-3ffb6fa3000 ---p 00000000 00:00 0
3ffb6fa3000-3ffb7044000 rw-p 00000000 00:00 0
3ffb7044000-3ffb7045000 ---p 00000000 00:00 0
3ffb7045000-3ffb70e6000 rw-p 00000000 00:00 0
3ffb70e6000-3ffb70e7000 ---p 00000000 00:00 0
3ffb70e7000-3ffb7188000 rw-p 00000000 00:00 0
3ffb7188000-3ffb7189000 ---p 00000000 00:00 0
3ffb7189000-3ffb722a000 rw-p 00000000 00:00 0
3ffb722a000-3ffb722b000 ---p 00000000 00:00 0
3ffb722b000-3ffb72cc000 rw-p 00000000 00:00 0
3ffb72cc000-3ffb72cd000 ---p 00000000 00:00 0
3ffb72cd000-3ffb736e000 rw-p 00000000 00:00 0
3ffb736e000-3ffb736f000 ---p 00000000 00:00 0
3ffb736f000-3ffb7410000 rw-p 00000000 00:00 0
3ffb7410000-3ffb7411000 ---p 00000000 00:00 0
3ffb7411000-3ffb74b2000 rw-p 00000000 00:00 0
3ffb74b2000-3ffb74b3000 ---p 00000000 00:00 0
3ffb74b3000-3ffb7554000 rw-p 00000000 00:00 0
3ffb7554000-3ffb7555000 ---p 00000000 00:00 0
3ffb7555000-3ffb9700000 rw-p 00000000 00:00 0
3ffb9700000-3ffb9702000 r-xp 00000000 fd:00 33727054                     /usr/lib64/libfreebl3.so
3ffb9702000-3ffb9703000 r--p 00001000 fd:00 33727054                     /usr/lib64/libfreebl3.so
3ffb9703000-3ffb9704000 rw-p 00002000 fd:00 33727054                     /usr/lib64/libfreebl3.so
3ffb9780000-3ffb992f000 r-xp 00000000 fd:00 33728058                     /usr/lib64/libc-2.17.so
3ffb992f000-3ffb9933000 r--p 001ae000 fd:00 33728058                     /usr/lib64/libc-2.17.so
3ffb9933000-3ffb9935000 rw-p 001b2000 fd:00 33728058                     /usr/lib64/libc-2.17.so
3ffb9935000-3ffb9939000 rw-p 00000000 00:00 0
3ffb9980000-3ffb9a28000 r-xp 00000000 fd:00 33728067                     /usr/lib64/libm-2.17.so
3ffb9a28000-3ffb9a29000 r--p 000a7000 fd:00 33728067                     /usr/lib64/libm-2.17.so
3ffb9a29000-3ffb9a2a000 rw-p 000a8000 fd:00 33728067                     /usr/lib64/libm-2.17.so
3ffb9a80000-3ffb9a8d000 r-xp 00000000 fd:00 33728062                     /usr/lib64/libcrypt-2.17.so
3ffb9a8d000-3ffb9a8e000 r--p 0000c000 fd:00 33728062                     /usr/lib64/libcrypt-2.17.so
3ffb9a8e000-3ffb9a8f000 rw-p 0000d000 fd:00 33728062                     /usr/lib64/libcrypt-2.17.so
3ffb9a8f000-3ffb9abd000 rw-p 00000000 00:00 0
3ffb9b00000-3ffb9b03000 r-xp 00000000 fd:00 33728065                     /usr/lib64/libdl-2.17.so
3ffb9b03000-3ffb9b04000 r--p 00002000 fd:00 33728065                     /usr/lib64/libdl-2.17.so
3ffb9b04000-3ffb9b05000 rw-p 00003000 fd:00 33728065                     /usr/lib64/libdl-2.17.so
3ffb9b80000-3ffb9b88000 r-xp 00000000 fd:00 35076903                     /usr/lib64/librt-2.17.so
3ffb9b88000-3ffb9b89000 r--p 00007000 fd:00 35076903                     /usr/lib64/librt-2.17.so
3ffb9b89000-3ffb9b8a000 rw-p 00008000 fd:00 35076903                     /usr/lib64/librt-2.17.so
3ffb9c00000-3ffb9c18000 r-xp 00000000 fd:00 35076899                     /usr/lib64/libpthread-2.17.so
3ffb9c18000-3ffb9c19000 r--p 00017000 fd:00 35076899                     /usr/lib64/libpthread-2.17.so
3ffb9c19000-3ffb9c1a000 rw-p 00018000 fd:00 35076899                     /usr/lib64/libpthread-2.17.so
3ffb9c1a000-3ffb9c1e000 rw-p 00000000 00:00 0
3ffb9c80000-3ffb9c97000 r-xp 00000000 fd:00 33728085                     /usr/lib64/libz.so.1.2.7
3ffb9c97000-3ffb9c98000 r--p 00016000 fd:00 33728085                     /usr/lib64/libz.so.1.2.7
3ffb9c98000-3ffb9c99000 rw-p 00017000 fd:00 33728085                     /usr/lib64/libz.so.1.2.7
3ffb9d00000-3ffb9d02000 r-xp 00000000 fd:00 268597581                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/encdb.so
3ffb9d02000-3ffb9d03000 r--p 00001000 fd:00 268597581                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/encdb.so
3ffb9d03000-3ffb9d04000 rw-p 00002000 fd:00 268597581                    /opt/rbenv/versions/2.7.3/lib/ruby/2.7.0/s390x-linux/enc/encdb.so
3ffb9d80000-3ffba175000 r-xp 00000000 fd:00 302040541                    /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7.3
3ffba175000-3ffba176000 ---p 003f5000 fd:00 302040541                    /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7.3
3ffba176000-3ffba17b000 r--p 003f5000 fd:00 302040541                    /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7.3
3ffba17b000-3ffba17f000 rw-p 003fa000 fd:00 302040541                    /opt/rbenv/versions/2.7.3/lib/libruby.so.2.7.3
3ffba17f000-3ffba198000 rw-p 00000000 00:00 0
3ffba19f000-3ffba200000 rw-p 00000000 00:00 0
3ffba200000-3ffba223000 r-xp 00000000 fd:00 33728047                     /usr/lib64/ld-2.17.so
3ffba224000-3ffba225000 r--p 00023000 fd:00 33728047                     /usr/lib64/ld-2.17.so
3ffba225000-3ffba226000 rw-p 00024000 fd:00 33728047                     /usr/lib64/ld-2.17.so
3ffba226000-3ffba227000 rw-p 00000000 00:00 0
3ffba274000-3ffba275000 rw-p 00000000 00:00 0
3ffba275000-3ffba27e000 rw-p 00000000 00:00 0
3ffba27e000-3ffba280000 r-xp 00000000 00:00 0                            [vdso]
3ffc2601000-3ffc2e00000 rw-p 00000000 00:00 0                            [stack]

Aborted (core dumped)
```
Reproduction script:
```
require 'google/protobuf'  # google-protobuf (3.11.0)
require 'base64'

Google::Protobuf::DescriptorPool.generated_pool.build do
  add_message "AsperaEvent" do
    optional :app_id, :string, 1
    optional :app_version, :string, 2
    optional :timestamp, :uint64, 3
    optional :customer_id, :string, 4
    optional :device_id, :string, 5
    optional :event_id, :string, 6
    optional :event_type, :string, 7
    map :event_data, :string, :message, 8, "AsperaEvent.EventData"
  end
  add_message "AsperaEvent.EventData" do
    oneof :data do
      optional :string_val, :string, 1
      optional :bytes_val, :bytes, 2
      optional :int_val, :int64, 3
      optional :float_val, :double, 4
      optional :bool_val, :bool, 5
    end
  end
end

AsperaEvent = Google::Protobuf::DescriptorPool.generated_pool.lookup("AsperaEvent").msgclass
AsperaEvent::EventData = Google::Protobuf::DescriptorPool.generated_pool.lookup("AsperaEvent.EventData").msgclass


message = Base64::decode64("
CgRhc2NwEgw0LjIuMC4xODM0NDMY0JGnv86ltcMWIgZhc3BlcmEqBGFzY3Ay
JGIzYWE4ZmZhLTI3NjQtNDkxZS1iN2U0LWJkN2RmYmYxYjYyNDoPYmxvY2tf
bGluZV9zZW5kQjQKCnNlc3Npb25faWQSJgokNjdhZjYxOGUtN2JjMi00Yzkx
LTk2N2ItOGMwZmU0ODllNTE4QjEKB3hmZXJfaWQSJgokNjdhZjYxOGUtN2Jj
Mi00YzkxLTk2N2ItOGMwZmU0ODllNTE4Qg4KCmFjY2Vzc19rZXkSAEITCglk
aXJlY3Rpb24SBgoEc2VuZEISCgxibG9ja3NfdG90YWwSAhgDQhUKD2Jsb2Nr
c19vcmlnaW5hbBICGAJCFAoOZHVwX2xhc3RfYmxvY2sSAhgBQhAKCmJsb2Nr
c19yZXgSAhgBQhYKEGxvd19jb250aWdfYmxvY2sSAhgAQhMKDW5leHRfcnhf
YmxvY2sSAhgAQhMKDWN0bHBkdV9iYWRtYWMSAhgAQhYKEGN0bHBkdV9iYWRz
ZXFudW0SAhgAQhUKD3JleF9ibG9ja3NfcmN2ZBICGAFCEwoNcmV4X3JlcXNf
cmN2ZBICGAFCDgoIcmJ0X3NpemUSAhgAQhcKEXJfb3R0X251bV9zYW1wbGVz
EgIYAUINCgdyX290dF9xEgIYAEIQCgpyX290dF9za2V3EgIYAEITCg1udW1f
eG1pdF9kb25lEgIYAUIXChB0YXJnZXRfcmF0ZV9rYnBzEgMYkE5CEwoNbWlu
X3JhdGVfa2JwcxICGABCFQoOY2FsY19yYXRlX2ticHMSAxiIDkIXChB0b3Rh
bF9ieXRlc19zZW50EgMYsAFCHAoVdG90YWxfZmlsZV9ieXRlc19zZW50EgMY
sAFCFAoMZWxhcHNlZF91c2VjEgQYyKAI
")

puts event = AsperaEvent.decode(message)
```
**Anything else we should know about your project / environment**
I have the core file, and can reproduce easily, to gather more details  if needed.

**Comment by fowles** [comment]
There have been a large number of fixes to ruby in the recent releases.  Does this still occur on the latest version, 3.17.3?

**Comment by pantras** [comment]
No the issue is fixed in google-protobuf 3.17.3. Thank you.
